pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -133,6 +133,47 @@ func changefeedTypeCheck(
 	return true, withSinkHeader, nil
 }
 
+func maybeShowCreateChangefeedWarning(
+	ctx context.Context, p sql.PlanHookState, opts changefeedbase.StatementOptions,
+) error {
+	st, err := opts.GetInitialScanType()
+	if err != nil {
+		return err
+	}
+
+	if opts.HasStartCursor() && st != changefeedbase.OnlyInitialScan {
+		statementTS := p.ExtendedEvalContext().GetStmtTimestamp().UnixNano()
+		cursorTS, err := evalTimestamp(ctx, p, hlc.Timestamp{WallTime: statementTS}, opts.GetCursor())
+		if err != nil {
+			return err
+		}
+
+		warningAge := int64(5 * time.Hour)
+		cursorAge := func() int64 {
+			knobs, _ := p.ExecCfg().DistSQLSrv.TestingKnobs.Changefeed.(*TestingKnobs)
+			if knobs != nil && knobs.OverrideCursorAge != nil {
+				return knobs.OverrideCursorAge()
+			}
+			return statementTS - cursorTS.WallTime
+		}()
+
+		if cursorAge > warningAge {
+			err = p.SendClientNotice(ctx,
+				pgnotice.Newf(
+					`the provided cursor is %d hours old; `+
+						`older cursors can result in increased changefeed latency`,
+					cursorAge/int64(time.Hour),
+				),
+				true,
+			)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return err
```

Comment:
nit: it makes it just a bit harder to read to return a variable when we know for sure it's nil

```suggestion
	return nil
```